cmake/list_insert_options.cmake

```diff
@@ -0,0 +1,55 @@
+#
+# Append key[=value] elements from one list to another list if they are not already in the target list.
```

Comment:
We should clarify the goal here first. Then come up with the behavior and name of this function.

The function is being used when calling `find_package(catkin REQUIRED COMPONENTS foo bar)` where as both components might have (recursive) dependencies themselves. What do you expect to happen if they export different values for the same key and why?

Re-reading some of this thread in the original comment you mentioned that this would be a "collision" and considered to print a warning.